### DISCUSSION TITLE
Jarred Sumner

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#96] [Announcements]
## Background:

[Twitter](https://twitter.com/jarredsumner)
[GitHub](https://github.com/Jarred-Sumner)
[Website](https://jarredsumner.com)
[bun](https://github.com/oven-sh/bun)

## Intro

Andrew: Hello, welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin. 

Justin: Hey everyone! Our guest today is Jarred Sumner. Jarred is the creator of Bun, a JavaScript runtime that also has a built in bundler, transpiler, task runner, and dependency manager! This is an incredibly ambitious project and the performance numbers coming out of Bun are incredibly exciting. Jarred, so great to have you on. Before we start talking about Bun, could you tell our audience a little more about you?

## Questions

- [x] You did the theil fellowship. For our listeners who might not know: What is that? What did you get take away from it?
- [x] What is bun? What other things is it comparable to?
- [x] Why did you initially start building bun? What was the initial problem you were trying to solve?
- [x] Bun seems to aim be quite an expansive tool, the scope is quite large. Why take on so much? Just implementing all of n_api seems like a big undertaking.
- [x] You're using (mostly) Zig to write Bun. Dino uses rust and Node primarily uses C/C++. What motivated you to use Zig and how's your experience been so far?
- [x] How pluggable do you plan to make bun? It has a transpiler, can I write code transforms? Will the bundler have plugins?
- [ ] What do you think is the most defining feature of bun that make it exciting to you?
- [ ] It seems like you've reimplemented much of javascript and node, what was the hardest thing to get right? 
- [x] How were you able to make things so fast? You seem to be a man on a mission to get those numbers down.
- [x] What's the end goal of bun? I've read some things about per-client bundling at the edge. Why is that something I'd want?
- [ ] Do you plan to make a company around bun? Do you want it to become more community driven?

## Tooltips

Add them in the comments below!

## Outro

Andrew: Well that’s it for this week’s episode of devtools.fm. Be sure to follow us on YouTube and wherever you consume your podcasts. Thanks for listening!

**Comment by hipstersmoothie** [comment]
My Tooltips:

- https://fork.dev
- https://github.com/player-ui/player

**Comment by just-be-dev** [comment]
Tooltips:
- https://blog.jgc.org/2022/07/guest-wifi-using-qr-code.html
- https://github.com/secure-77/Perlite

**Comment by hipstersmoothie** [comment]
Jarred's tooltips;

https://github.com/apvarun/awesome-bun
